### PR TITLE
Make devices and presences lack property names

### LINKED ISSUE #792: Presences should not have ownProperties
Currently, if you pass an empty object across SwingSet vats, it is marshalled as a Presence (or a Device), and has a `.toString` ownProperty added to it. This makes for a surprise on the other end, where an ownProperty is not expected.  This has bitten us in several cases where an empty object is a desirable feature of an interface that takes 0 or more ownProperties.

If instead we set the presence's prototype to have the .toString then it won't have any .keys() or .getOwnPropertyNames(), so it will be handled correctly as an empty object!

@warner I'll work out a PR for this.

### PR DESCRIPTION
**Pull request by michaelfig**
Closes #792 

This PR improves the debuggability and predictability of device nodes and presences created by SwingSet and CapTP.  Adding a `Symbol.for('SwingSet')` property (which doesn't show up with `Object.getOwnPropertyNames`), and setting a prototype that defines `toString()` (which allows the presence to be stringified as before) means that empty objects are marshalled compatibly as empty presences.

The REPL is improved to sniff specifically for this mechanism to distinguish presences from empty objects.

### CODE REVIEW

**Comment by warner** [comment]
Ugh, `tap` is starting to annoy me.. it's really hard to find where the test failure is.

Looks like it's in SwingSet, in `test-marshal.js`. I'm guessing we have a test that expects `.toString` to be present.

**Comment by michaelfig** [comment]
I've updated the approach: the current idea is to represent presences as an empty object with a prototype that defines:

```js
{
  toString() { ... },
  [Symbol.for('SwingSet')]: true,
  [Symbol.toStringTag]: description,
}
```

This allows for simple identification of the objects in the REPL, as well as informative printing and stringification.

**Comment by michaelfig** [comment]
For example:

```
> const o = ...made as described;
> o
Object [Presence 52] {}
> String(o)
'[Presence 52]'
> throw o
Uncaught Object [Presence 52] {}
> Object.getOwnPropertyNames(o)
[]
> Object.getOwnPropertySymbols(o)
[]
>
```

File: packages/cosmic-swingset/lib/ag-solo/vats/repl.js
Comment (warner) [inline packages/cosmic-swingset/lib/ag-solo/vats/repl.js:4]: I'm not sure why, but I'm a tiny bit uncomfortable about expanding the contract between liveSlots, and the javascript environment it hosts. There a little bit of magic here that we'll have to include in our definition of Jessie or "the swingset javascript environment" or whatever we end up calling the userspace of a vat.

But I don't think that's a reason to not do this.

**Review by warner — APPROVED** [review]
If we're going to implement #792, this is a reasonable way to do it. But please get @erights 's approval on the overall approach before landing, he's got a better sense of what sorts of promise we ought to be making than I do.

File: packages/cosmic-swingset/lib/ag-solo/vats/repl.js
Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/vats/repl.js:4]: Jessie doesn't have symbols, BTW.  And if the contract is broken, all that happens is Presences aren't easily detectable by the "swingset javascript environment".

Comment (erights) [reply packages/cosmic-swingset/lib/ag-solo/vats/repl.js:4]: Jessie not having symbols seems like a problem with this approach. So far, everything passable is a subset of what can be represented and manipulated from Jessie.

Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/vats/repl.js:4]: The symbol is not passable or passed.  It's between the liveSlots interface and user code, all of which lives in the same vat.  So, the "detect if it's a presence" code cannot be written within Jessie, but I think she need not concern herself with it.

Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/vats/repl.js:4]: Maybe we need `HandledPromise.isPresence(p)` ? If we can agree on that, then there's no need for the special symbol usage (the REPL can use `isPresence`), and this PR reduces to just ensuring presences have no ownProperties, as originally desired.

Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/vats/repl.js:4]: @erights and I have agreed upon #804.  I will be implementing the first stage of it in this PR.

**Comment by erights** [comment]
Is this subsumed by #804 ?

**Comment by michaelfig** [comment]
> Is this subsumed by #804 ?

Yes.

**Pull request closed without merging**
